<launch>  
	<arg name="rgb_image_node" default="/zed2/zed_node/rgb/image_rect_color"/>
	<arg name="depth_image_node" default="/zed2/zed_node/depth/depth_registered"/>
	<arg name="zed_params_dir" default="$(find object_slam)/config"/>
	<arg name="zed_params" default="/RGB-D/ZED2.yaml"/>


	<!-- SLAM  -->
	<node pkg="object_slam" name="zed_rgbd_node" type="zed_rgbd_node"
		  args="/home/lacie/slam_ws/src/junbot_planner/object_slam/object_slam/Vocabulary/ORBvoc.txt.bin
	            /home/lacie/slam_ws/src/junbot_planner/object_slam/config/RGB-D/ZED2.yaml" output="screen">

		<!-- Parameters for original ORB-SLAM3 -->
    	<param name="voc_file"      type="string" value="$(find object_slam)/object_slam/Vocabulary/ORBvoc.txt.bin"/>
    	<param name="settings_file" type="string" value="$(find object_slam)/config/RGB-D/ZED2.yaml"/>

		<remap from="/camera/rgb/image_raw"                 to="/zed2/zed_node/rgb/image_rect_color"/>
    	<remap from="/camera/depth_registered/image_raw"    to="/zed2/zed_node/depth/depth_registered"/>

		<!-- Parameters for ROS -->
    	<param name="worl4d_frame_id"    type="string"   value="world" />
    	<param name="cam_frame_id"      type="string"   value="camera" />
		<param name="enable_pangolin"   type="bool"     value="true" />
		
	</node>

	<!-- Run other launch file -->
    <include file="$(find object_slam)/launch/octomap.launch" />
	<include file="$(find object_slam)/launch/transform.launch" />
	<node pkg="rviz" name="rviz" type="rviz" /> 
	
</launch>
